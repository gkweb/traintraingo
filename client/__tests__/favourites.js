import React from 'react'
import 'jest-styled-components'
import { shallow, mount } from 'enzyme'
import Favourites from './../components/favourites'
import waitForExpect from 'wait-for-expect'

// Stops Mock
const mockData = {
  '1063': 'Epping Station',
  '1108': 'Kensington Station',
}

test('Favourites display when mock data is set (No local storage for unit test)', () => {
  const c = mount(<Favourites />)

  c.setState({
    favourites: mockData,
  })

  c.update()

  waitForExpect(() => expect(c.text()).toContain('Epping Station'))
})

test('Favourites Snapshot', () => {
  const c = shallow(<Favourites />)
  expect(c).toMatchSnapshot()
})
