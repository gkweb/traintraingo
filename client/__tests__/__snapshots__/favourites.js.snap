// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favourites Snapshot 1`] = `""`;
